pkg/node/node.go

```diff
@@ -454,27 +455,57 @@ func NewBee(interrupt chan struct{}, addr string, publicKey *ecdsa.PublicKey, si
 		return nil, err
 	}
 
-	var (
-		blockHash []byte
-		txHash    []byte
-	)
+	// if theres a previous transaction hash, and not a new chequebook deployment on a node starting from scratch
+	// get old overlay
+	// mine nonce that gives similar new overlay
+	nonce, nonceExists, err := overlayNonceExists(stateStore)
+	existingOverlay, err := GetExistingOverlay(stateStore)
 
-	txHash, err = GetTxHash(stateStore, logger, o.Transaction)
 	if err != nil {
```

Comment:
This error is being overridden. We would need to check the error in `overlayNonceExists` separately right?

If there is no nonce, do we still need to GetExistingOverlay?

Reply:
fixed